trace/tracestate.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"container/list"
+	"errors"
+	"fmt"
+)
+
+const (
+	// KeyMaxSize is the maximum characters allowed in the key.
+	KeyMaxSize = 256
+
+	// ValueMaxSize is the maximum characters allowed in the value.
+	ValueMaxSize = 256
+
+	// MaxKeyValuePairs is the maximum number of key-value pairs allowed in the tracestate.
+	MaxKeyValuePairs = 32
+)
+
+// Tracestate represent tracing-system specific context in a list of key-value pairs. Tracestate allows different
+// vendors propagate additional information and inter-operate with their legacy Id formats.
+type Tracestate struct {
+	stateList *list.List
```

Comment:
Don't use the container package, it is not maintained by anyone.

Why do you need a list?